### PR TITLE
add testnet support

### PR DESCRIPTION
**Pull request by aardvark3**
A selector view is added to switch to the test net, when the wallet is empty.

### CODE REVIEW

File: bitherj/src/main/java/net/bither/bitherj/PrimerjSettings.java [hunk 1]
```diff
@@ -54,13 +54,19 @@
 
     public static final BigInteger proofOfWorkLimit = Utils.decodeCompactBits(0x1d00ffffL);
     public static final int port = 9911;
+    public static final int testNetPort = 9913;
+    public static int getPort(boolean isTestNet){
+        if(isTestNet) return testNetPort;
+        return port;
+    }
     //    public static final int port = 8333;
     public static final long packetMagic = 0xe4e7e5e7L;
```
Comment (sunnyking) [inline bitherj/src/main/java/net/bither/bitherj/PrimerjSettings.java:63]: iirc the magic is different for the testnet as well

File: bitherj/src/main/java/net/bither/bitherj/PrimerjSettings.java [hunk 2]
```diff
@@ -74,6 +80,11 @@
     public static final int btfP2shHeader = 40;
     public static final int btpP2shHeader = 58;
 
+    public static int getAddressHeader(boolean isTestNet) {
+        if(isTestNet) return testAddressHeader;
```
Comment (sunnyking) [inline bitherj/src/main/java/net/bither/bitherj/PrimerjSettings.java]: i see you always call utils for the isTestNet, maybe better to call utils here instead so caller doesn't need to

File: bitherj/src/main/java/net/bither/bitherj/api/http/PrimerUrl.java [hunk 3]
```diff
@@ -54,6 +54,16 @@
 
     public static final String GET_BY_ADDRESS = "https://explorer.primecoin.net/api/searchrawtransactions/%s";
     public static final String GET_BY_SYNCBLOCK = "https://explorer.primecoin.net/api/syncblock/";
+    public static final String GET_BY_ADDRESS_TESTNET = "https://testexplorer.primecoin.net/api/searchrawtransactions/%s";
+    public static final String GET_BY_SYNCBLOCK_TESTNET = "https://testexplorer.primecoin.net/api/syncblock/";
+    public static String getByAddress(boolean isTestNet){
```
Comment (sunnyking) [inline bitherj/src/main/java/net/bither/bitherj/api/http/PrimerUrl.java]: same here, appears no need to pass in isTestNet

File: bitherj/src/main/java/net/bither/bitherj/api/http/PrimerUrl.java [hunk 4]
```diff
@@ -54,6 +54,16 @@
...
+        if(isTestNet) return GET_BY_ADDRESS_TESTNET;
+        return GET_BY_ADDRESS;
+    }
+    public static String getBySyncBlock(boolean isTestNet){
```
Comment (sunnyking) [inline bitherj/src/main/java/net/bither/bitherj/api/http/PrimerUrl.java]: also here

File: bitherj/src/main/java/net/bither/bitherj/api/http/PrimerUrl.java [hunk 5]
```diff
@@ -97,6 +107,11 @@
 
     //other
     public static final String BCHAIN_INFO_ADDRESS_URL = "https://explorer.primecoin.net/address/";
+    public static final String BCHAIN_INFO_ADDRESS_URL_TESTNET = "https://testexplorer.primecoin.net/address/";
+    public static String bChainInfoAddressUrl(boolean isTestNet){
```
Comment (sunnyking) [inline bitherj/src/main/java/net/bither/bitherj/api/http/PrimerUrl.java]: also here

File: bitherj/src/main/java/net/bither/bitherj/PrimerjSettings.java (hunk 1 above)
Comment (aardvark3) [reply bitherj/src/main/java/net/bither/bitherj/PrimerjSettings.java:63]: Yes, for the testnet it is 0xfbfecbc3L. Thanks for this hint.

File: bitherj/src/main/java/net/bither/bitherj/PrimerjSettings.java (hunk 2 above)
Comment (aardvark3) [reply bitherj/src/main/java/net/bither/bitherj/PrimerjSettings.java]: Good suggestion, I have modified this

File: bitherj/src/main/java/net/bither/bitherj/PrimerjSettings.java [hunk 6]
```diff
@@ -54,13 +54,20 @@
...
+    }
     //    public static final int port = 8333;
     public static final long packetMagic = 0xe4e7e5e7L;
+    public static final long testNetPacketMagic = 0xfbfecbc3L;
 //    public static final long packetMagic = 0xf9beb4d9L;
 
 
     //    public static final int addressHeader = 0;
     public static final int addressHeader = 23;
+    public static final int testAddressHeader = 111;
```
Comment (sunnyking) [inline bitherj/src/main/java/net/bither/bitherj/PrimerjSettings.java]: To be consistent with other names, this best to be 'testNetAddressHeader'

Comment (aardvark3) [reply bitherj/src/main/java/net/bither/bitherj/PrimerjSettings.java]: Ok,

**Review by sunnyking — APPROVED** [review]

**Pull request merged**